aws/monitorinstances.go

```diff
@@ -0,0 +1,308 @@
+		res = append(res, aws.StringValue(region.RegionName))
+	}
+	return res
+}
+
+// getPropertiesMapping uses the reflection of the struct InstanceInfo and
+// InstanceTag to get the mapping of the ElasticSearch documents.
+func getPropertiesMapping(field interface{}) map[string]interface{} {
+	properties := make(map[string]interface{})
+	rtype := reflect.TypeOf(field)
+	for i := 0; i < rtype.NumField(); i++ {
+		name := strings.ToLower(rtype.Field(i).Name)
+		if value, ok := rtype.Field(i).Tag.Lookup("json"); ok {
+			name = value
+		}
+		properties[name] = make(map[string]interface{})
+		for _, tagAvailable := range tagsAvailable {
+			if value, ok := rtype.Field(i).Tag.Lookup(tagAvailable); ok {
+				properties[name].(map[string]interface{})[tagAvailable] = value
+			}
+		}
+	}
+	if _, ok := rtype.FieldByName("Tags"); ok {
+		properties["tags"] = map[string]interface{}{
+			"properties": getPropertiesMapping(InstanceTag{}),
+		}
+	}
+	return properties
+}
+
+// createIndexES creates the ElasticSearch index with the properties
+// from getPropertiesMapping.
+func createIndexES(client *elastic.Client, logger jsonlog.Logger) {
+	properties := getPropertiesMapping(InstanceInfo{})
+	mapping := map[string]interface{}{
+		"mappings": map[string]interface{}{
+			elasticsearchType: map[string]interface{}{
+				"properties": properties,
+			},
+		},
+	}
+	if _, err := client.CreateIndex(elasticsearchIndex).BodyJson(mapping).Do(context.Background()); err != nil {
+		logger.Error("ElasticSearch create index failed", err)
+		return
+	}
+}
+
+// importInstancesToES imports an array of InstanceInfo in ElasticSearch.
+// It calls createIndexES if the index doesn't exist.
+func importInstancesToES(instances []InstanceInfo, logger jsonlog.Logger) {
+	// TODO: Get credentials
+	client, err := es.NewSignedElasticClient(config.EsUrl, credentials.NewSharedCredentials("", "default"))
+	if err != nil {
+		logger.Error("Get ElasticSearch client failed", err)
+		return
+	}
+	if exists, err := client.IndexExists(elasticsearchIndex).Do(context.Background()); err != nil {
+		logger.Error("ElasticSearch check if index exists failed", err)
+		return
+	} else if !exists {
+		createIndexES(client, logger)
+	}
+	for _, instance := range instances {
+		if res, err := client.Index().Index(elasticsearchIndex).Type(elasticsearchType).BodyJson(instance).Do(context.Background()); err != nil {
+			logger.Error("Instance put in ES failed", err)
+		} else {
+			logger.Info("Instance put in ES", *res)
+		}
+	}
+}
+
+// FetchInstancesStats fetchs the stats of the EC2 instances of an AwsAccount
+// to import them in ElasticSearch. The stats are fetched from the last hour.
+// In this way, FetchInstancesStats should be called every hour.
+func FetchInstancesStats(awsAccount AwsAccount) {
+	logger := jsonlog.DefaultLogger
+	tcreds, err := GetTemporaryCredentials(context.Background(), awsAccount, "fetchInstanceStats")
+	creds := credentials.NewStaticCredentials(
```

Comment:
Using STS credentials this way means your function will start failing after one hour. Please look at function `getServiceForRepository` in file _/aws/s3/readBills.go_ for a correct example.